presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -162,27 +161,37 @@ public void addInput(Page page)
     @Override
     public Page getOutput()
     {
-        if (lookupSource == null) {
+        if (probe == null && pageBuilder.isEmpty()) {
+            // Fast exit path when lookup source is still being build
             return null;
         }
 
+        checkState(lookupSourceProvider != null, "Input has been accepted before lookup source provided");
+
+        try (LookupSourceProvider.LookupSourceLease lookupSourceLease = lookupSourceProvider.leaseLookupSource()) {
```

Comment:
Static import. 
I see some later commit does that but it should be done already here.